src/cmds/wc.c

```diff
@@ -0,0 +1,252 @@
+     
+	if (!file_in) {
+		printf("Error with opening file: %s\n", name);
+		exit(1);
+	}
+     
+	while ((byte_file = fgetc(file_in)) != EOF) {      
+		if (byte_file != '\n') {
+			if (byte_file == '\t') {
+				count = count + 8; 
+
+			} else {
+				count++;
+			}
+
+		} else {
+
+		if (max_len < count) {
+			max_len = count;
+		}
+		count = 0;
+		}
+	}
+
+	fclose(file_in);
+     
+	return max_len;
+}
+
+static int number_of_bytes(char *name) {
+	int   byte_file = 0;
+	int   count = 0;
+     
+	FILE*    file_in = fopen(name, "rt");
+     
+	if (!file_in) {
+		printf("Error with opening file: %s\n", name);
+		exit(1);
+	}
+     
+	while ((byte_file = fgetc(file_in)) != EOF) {      
+		count++;
+	}
+     
+	fclose(file_in);
+     
+	return count;
+}
+
+static int number_of_words(char *name) {
+	int   byte_file = 0;
+	int   count = 0;
+     
+	bool  prev_symbol = false;
+	bool  next_symbol = false;
+     
+	FILE*    file_in = fopen(name, "rt");
+     
+	if (!file_in) {
+		printf("Error with opening file: %s\n", name);
+		exit(1);
+	}
+     
+	while ((byte_file = fgetc(file_in)) != EOF) {      
+		next_symbol = !isspace(byte_file)  ?  true  :  false;
+     
+		if (next_symbol) {
+             		prev_symbol = true;
+          	}
+     
+          	if (prev_symbol && !next_symbol){
+                	prev_symbol = false;
+    			count++;
+          	}
+	}
+     
+	fclose(file_in);
+     
+	return count;
+}
+
+static int number_of_strings(char *name) {
+	int   byte_file = 0;
+	int   prev_byte = 0;
+	int   count = 0;
+	int empty_file = true;
+	
+	FILE*    file_in = fopen(name, "rt");
+ 
+	if (!file_in) {
+		printf("Error with opening file: %s\n", name);
+		exit(1);
+	}
+
+ 	while ((byte_file = fgetc(file_in)) != EOF) { 
+		prev_byte = byte_file; 
+ 		
+		empty_file = false;
+        	if (byte_file == '\n') { 
+            		count++; 
+		}
+	}
+
+	if (empty_file) { 
+        	return 0; 
+	} else if (prev_byte != '\n') { 
+		count++; 
+	}
+   
+        return count;
+}
+
+int main(int argc, char **argv) { 
+	char *file_name;
+	int opt, param_number;
+	param_number=0;
+
+	static bool print_lines, print_words,print_bytes;
+	static bool print_linelength;
+	
+	print_lines = print_words = print_bytes = false;
+  	print_linelength = false;
+
+	if (argc == 0) {
+		return -EINVAL;
+	}
+
+	if (argv[1] == NULL) {
+		printf("You need to specify the parameter(file)\n");
+		return 0;
+	}
+
+	
+	while (-1 != (opt = getopt(argc, argv, "lLcw"))) {
+		switch(opt) {
+			case 'l':
+			print_lines = true;
+			param_number++;
+			break;
+
+			case 'L':
+			print_linelength = true;
+			param_number++;
+			break;
+                        
+			case 'c':
+			print_bytes = true;
+			param_number++;
+			break;
+
+			case 'w':
+			print_words = true;
+			param_number++;
+			break;
+			
+		}
+	}
+
+	
+	
+	if (! (print_lines || print_words || print_bytes
+         || print_linelength))
+		print_lines = print_words = print_bytes = true;
+
+
+
+	
+	if (argc-param_number-1 == 1) {		 
```

Comment:
this case is really required? whether If argc > 1 the loop from else branch show all required parameters or not .